src/greynirseq/nicenlp/tasks/pos_task.py

```diff
@@ -17,84 +17,31 @@
     SortDataset,
     TruncateDataset,
     PrependTokenDataset,
+    ListDataset,
 )
 from fairseq.data import encoders
 from fairseq.tasks import FairseqTask, register_task
-from fairseq.tasks.sentence_prediction import SentencePredictionTask
-
-try:
-    from icecream import ic
-
-    ic.configureOutput(includeContext=True)
-except ImportError:  # Graceful fallback if IceCream isn't installed.
-    ic = lambda *a: None if not a else (a[0] if len(a) == 1 else a)  # noqa
 
 from greynirseq.nicenlp.data.datasets import (
     POSDataset,
     WordEndMaskDataset,
-    GroupMaskDataset,
     RightPad2dDataset,
-    DynamicLabelledSpanDataset,
-    LabelledSpanDataset,
-    SpanDataset,
-    SparseProductSpanDataset,
-    ProductSpanDataset,
-    NumSpanDataset,
     NestedDictionaryDatasetFix,
-    LossMaskDataset,
     NumWordsDataset,
 )
-import greynirseq.nicenlp.utils.greynir.greynir_utils as greynir_utils
+from greynirseq.nicenlp.utils.label_schema.label_schema import (
+    label_schema_as_dictionary,
+    parse_label_schema,
+    make_dict_idx_to_vec_idx,
+    make_vec_idx_to_dict_idx,
+    make_group_name_to_group_attr_vec_idxs,
+    make_group_masks,
+)
 
-logger = logging.getLogger(__name__)
 
+from greynirseq.nicenlp.utils.constituency import token_utils
 
-def parse_label_schema(path):
-    LabelSchema = namedtuple(
-        "LabelSchema",
-        [
-            "labels",
-            "group_name_to_labels",
-            "label_categories",
-            "category_to_group_names",
-            "separator",
-            "group_names",
-            "null",
-            "null_leaf",
-        ],
-    )
-    with open(path, "r") as fp:
-        j_obj = json.load(fp)
-    return LabelSchema(**j_obj)
-
-
-def label_schema_as_dictionary(label_schema):
-    label_dict = Dictionary()
-
-    labels = list(label_schema.labels)
-    assert len(labels) == len(set(labels))
-
-    for label in labels:
-        label_dict.add_symbol(label)
-
-    return label_dict
-
-
-def make_group_masks(schema, dictionary, device="cpu"):
-    num_groups = len(schema.group_names)
-    label_shift = dictionary.nspecial
-    num_labels = len(dictionary) - label_shift
-    ret_mask = torch.zeros(num_labels, num_groups, dtype=torch.int64, device=device)
-    for cat, cat_group_names in schema.category_to_group_names.items():
-        cat_label_idx = dictionary.index(cat)
-        cat_vec_idx = schema.label_categories.index(cat)
-        for group_name in cat_group_names:
-            ret_mask[cat_vec_idx, schema.group_names.index(group_name)] = 1
-        assert cat_label_idx != dictionary.unk()
-    for cat in schema.label_categories:
-        cat_label_idx = dictionary.index(cat)
-        assert cat_label_idx != dictionary.unk()
-    return ret_mask
+logger = logging.getLogger(__name__)
 
 
 @register_task("pos_ice")
```

Comment:
Nothing pos specific about this task?

Reply:
Agreed, renaming and changing filename as well.